topic-operator/pom.xml

```diff
@@ -22,6 +22,36 @@
             <groupId>io.strimzi</groupId>
             <artifactId>api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-annotations</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>io.kroxylicious.testing</groupId>
+            <artifactId>testing-api</artifactId>
+            <version>0.3.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.kroxylicious.testing</groupId>
+            <artifactId>testing-junit5-extension</artifactId>
+            <version>0.3.0</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>io.kroxylicious.testing</groupId>
+            <artifactId>testing-impl</artifactId>
+            <version>0.3.0</version>
+            <scope>test</scope>
+        </dependency>
```

Comment:
I would really rather not switch. I'm using some features of Kroxy testing which are not available in our own test containers. 

1. Most of the tests in the IT are using an in-VM cluster rather than using test containers, which means all ~200 tests run in about 2 minutes on my laptop. It would take much longer to run these tests if they were all using test containers, which would slow the development inner loop.
2. Kroxy testing provides a common abstraction over clusters which are in-VM and clusters which are running in containers, which is also a feature I'm using. It's actually very nice to be able to easily switch between the two when you realise that you need to resort to using containers.

In my opinion switching would require a lot of effort to rewrite very large test, which would then run slower for the rest of time, but for what actual benefit?

Reply:
I think we have a very bad experience with dependencies like that - e.g. the Debezium Kafka test clusters etc.

Reply:
What was the basis of the bad experience, exactly?

Reply:
* Almost all our dependencies are from outside projects which we cannot fix ourselves.
* How is a project to become well proven no one uses it? It's useful and it has an acceptable license. 
* The project in question has gone to the trouble of factoring this out as a separate dependency precisely for other projects to use (unlike DBZ ever did). They have specifically released this dependency to maven central. They've been making [regular releases](https://central.sonatype.com/artifact/io.kroxylicious.testing/testing-junit5-extension/0.3.0/versions). They're just about to release a new version with support for 3.5.0. 
* I don't think it's fair to use historical problems with a dependency from one project (Debezium) to make assertions about a dependency from another _completely different_ project. 

Finally, I wrote a lot of the code in this dependency, and work with the community in question, _unlike most of our other dependencies_. While that certainly makes me somewhat biased, it is also reason to doubt that this _really_ represents as much of a problem as you're suggesting.

Reply:
@ppatierno @scholzj @tombentley What are the actual risks of using kroxylicious? What's the worst case scenario if it goes pear shaped? I can't see any negative effects beyond ongoing effort to handle flakiness, or worst case, rewriting tests that use it. 

Tom states it offers iteration speed and additional features over test containers, faster iteration (red/green etc.) is very compelling in a testing tool, and tbh, the Strimzi tests built around test containers were rather brittle when I last looked at them.

With regards to the risk of needing ongoing effort to handle any flakiness, there's alway been an eternal implicit social contract among programmers when introducing new technologies - you brought it in, you're the one that people will expect to support and fix it. If Tom thinks the benefits of kroxylicious are worth shouldering that responsibility, then I trust his decision and his commitment to that implicit contract.

In short, there seems to be no evidence of actual risk, just hypotheticals and maybes, and I'd suggest not letting the perfect be the enemy of the good.

Reply:
> In short, there seems to be no evidence of actual risk, just hypotheticals and maybes, and I'd suggest not letting the perfect be the enemy of the good.

Seriously? Referring to actual past issues is `hypotheticals and maybes`. But some _wanna be funny_ references to non-existing social contracts are the real deal?

Reply:
@scholzj - actual past issues yep, but with a different library, extrapolating from bad experiences with Debezium to kroxylicious is what I deem hypothetical. Subjective, of course, so agree to disagree.

And wasn't trying for humour, Kiwi humour is dry, but it's not _that_ dry, that was my rather long-winded way of saying that "Tom isn't going to drop this dependency into the codebase via a drive-by PR and not support it",  and as such, the risk level is low.